aws-mfa

```diff
@@ -89,34 +85,41 @@ def validate(args, config):
         key_id = config.get(long_term_name, 'aws_access_key_id')
         access_key = config.get(long_term_name, 'aws_secret_access_key')
     except NoSectionError:
-        logger.error(
+        log_error_and_exit(
             "Long term credentials session '[%s]' is missing. "
             "You must add this section to your credentials file "
             "along with your long term 'aws_access_key_id' and "
             "'aws_secret_access_key'" % (long_term_name,))
-        sys.exit()
     except NoOptionError as e:
-        logger.error(e)
-        sys.exit()
+        log_error_and_exit(e)
 
     if not args.device:
         if os.environ.get('MFA_DEVICE'):
             args.device = os.environ.get('MFA_DEVICE')
         elif config.has_option(long_term_name, 'aws_mfa_device'):
             args.device = config.get(long_term_name, 'aws_mfa_device')
         else:
-            sys.exit('You must provide --device or MFA_DEVICE or set '
-                     '"aws_mfa_device" in ".aws/credentials"')
+            log_error_and_exit(
+                'You must provide --device or MFA_DEVICE or set '
+                '"aws_mfa_device" in ".aws/credentials"')
+
+    if not args.assume_role:
+        if os.environ.get('MFA_ASSUME_ROLE'):
+            args.assume_role = os.environ.get('MFA_ASSUME_ROLE')
 
     if not args.duration:
         if os.environ.get('MFA_STS_DURATION'):
             args.duration = int(os.environ.get('MFA_STS_DURATION'))
+    if args.duration:
```

Comment:
I think we should still define the default just as before this PR, but the only difference now is that we are being a bit more intelligent about it rather than just always choosing the minimum of 900 seconds. As for validation, if a user is uneducated and provides an invalid value, AWS will return a detailed exception of why it's invalid and what they can do to correct it.